pkg/virt-controller/watch/export/export.go

```diff
@@ -879,3 +930,128 @@ func (ctrl *VMExportController) pvcConditionFromPVC(pvcs []*corev1.PersistentVol
 	}
 	return cond
 }
+
+func (ctrl *VMExportController) getExternalLinkHostAndCert() (string, string) {
+	for _, obj := range ctrl.IngressCache.List() {
+		if ingress, ok := obj.(*networkingv1.Ingress); ok {
+			if host := getHostFromIngress(ingress); host != "" {
+				cert, _ := ctrl.getIngressCert(getDomainFromHost(host, ctrl.KubevirtNamespace), ingress)
+				return host, cert
+			}
+		}
+	}
+	for _, obj := range ctrl.RouteCache.List() {
+		if route, ok := obj.(*routev1.Route); ok {
+			if host := getHostFromRoute(route); host != "" {
+				cert, _ := ctrl.getRouteCert(getDomainFromHost(host, ctrl.KubevirtNamespace))
+				return host, cert
+			}
+		}
+	}
+	return "", ""
+}
+
+func (ctrl *VMExportController) getIngressCert(domainName string, ing *networkingv1.Ingress) (string, error) {
+	secretName := ""
+	for _, tls := range ing.Spec.TLS {
+		if tls.SecretName != "" {
+			secretName = tls.SecretName
+			break
+		}
+	}
+	key := controller.NamespacedKey(ctrl.KubevirtNamespace, secretName)
+	obj, exists, err := ctrl.SecretInformer.GetStore().GetByKey(key)
+	if err != nil {
+		return "", err
+	} else if !exists {
+		return "", nil
+	}
+	if secret, ok := obj.(*corev1.Secret); ok {
+		return ctrl.getIngressCertFromSecret(secret, domainName)
+	}
+	return "", nil
+}
+
+func (ctrl *VMExportController) getIngressCertFromSecret(secret *corev1.Secret, domainName string) (string, error) {
+	certBytes := secret.Data["tls.crt"]
+	certs, err := cert.ParseCertsPEM(certBytes)
+	if err != nil {
+		return "", err
+	}
+	return findCertByDomainName(domainName, certs)
+}
+
+func (ctrl *VMExportController) getRouteCert(domainName string) (string, error) {
+	key := controller.NamespacedKey(ctrl.KubevirtNamespace, routeCAConfigMapName)
+	obj, exists, err := ctrl.RouteConfigMapInformer.GetStore().GetByKey(key)
+	if err != nil {
+		return "", err
+	} else if !exists {
+		return "", nil
+	}
+	if cm, ok := obj.(*corev1.ConfigMap); ok {
+		cmString := cm.Data[routeCaKey]
+		certs, err := cert.ParseCertsPEM([]byte(cmString))
+		if err != nil {
+			return "", err
+		}
+		return findCertByDomainName(domainName, certs)
+	}
+	return "", fmt.Errorf("not a config map")
+}
+
+func findCertByDomainName(domainName string, certs []*x509.Certificate) (string, error) {
+	for _, cert := range certs {
+		if strings.Contains(cert.Subject.CommonName, domainName) {
+			return buildPemFromCert(cert), nil
+		}
+		for _, extension := range cert.Extensions {
+			if extension.Id.String() == subjectAltNameId && strings.Contains(string(extension.Value), domainName) {
```

Comment:
I'm not saying this is wrong, but I wonder if `contains` is a little too loose?  What are some real world examples?

Reply:
Example from CRC:
```
            X509v3 Subject Alternative Name: 
                DNS:*.apps-crc.testing
```